lib/api/Cargo.toml

```diff
@@ -14,7 +14,7 @@ tracing = ["dep:tracing", "segment/tracing"]
 [dependencies]
 tonic = { version = "0.9.2", features = ["gzip", "tls"] }
 prost = "0.11.9"
-prost-types = "0.11.9"
+prost-wkt-types = "0.4.2"
```

Comment:
Do we strictly require this? Would it be possible to implement it in a different way somehow?

We already have some problems with `prost` being slow to update, causing a bunch of other problems. I imagine this to only take longer when using a dependency like this.

Reply:
I did switch to `prost_wkt_types::Timestamp` because of validator and serde support, as requested in https://github.com/qdrant/qdrant/pull/3395#discussion_r1452756856.

It's not possible to solve it using annotations over `prost_types::Timestamp` due to [this issue](https://github.com/Keats/validator/issues/285). However, according to a maintainer, it will be fixed in next versions.

---

> We already have some problems with prost being slow to update

Could you elaborate, please? BTW, I did use old version of `prost-wkt-types` because it's compatible with an old version of `prost` that we use.

Reply:
Thanks for elaborating and linking to the sources.

I'd be fine to continue with it then to not make this more difficult. I guess we can switch back to regular `prost` once the necessary bits are in place.

> Could you elaborate, please? BTW, I did use old version of `prost-wkt-types` because it's compatible with an old version of `prost` that we use.

Yes. On second look, it's not actually prost being the problem here, but other dependencies around it that we use. Still, I'd prefer to minimize our dependency hell around prost. 
For example, see the comments here: https://github.com/qdrant/qdrant/pull/2589

But, lets leave it like this for now :+1: